File: node/src/api/middleware/authMiddleware.js
const admin = require('firebase-admin');

exports.checkAuth = async (req, res, next) => {
    try {
        const token = req.headers.authorization.split('Bearer ')[1];
        const decodedToken = await admin.auth().verifyIdToken(token);
        req.user = decodedToken;
        req.user = { email: decodedToken.email, uid: decodedToken.uid };
        next();
    } catch (error) {
        res.status(401).json({ message: 'Unauthorized' });
    }
};

File: fastapi/app/middleware/auth_middleware.py
from fastapi import Request, HTTPException
from firebase_admin import auth

async def check_auth(request: Request, call_next):
    token = request.headers.get('Authorization')
    try:
        decoded_token = auth.verify_id_token(token)
        request.state.user = decoded_token
    except Exception as e:
        raise HTTPException(status_code=401, detail="Invalid token")
    
    return await call_next(request)
